File: python/exercise_234.py
#fazer em O(n) time e O(1) space depois usando fast and slow pointer
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        lista = []
        headAux = head
        count = 0
        while headAux != None:
            lista.insert(0,headAux.val)
            headAux = headAux.next
        for i in range(int(len(lista)/2)):
            if head.val != lista[i]:
                return False
            head = head.next
        return True

File: java/linkedList/exercise_234.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean isPalindrome(ListNode head) {
        List<Integer> lista = new ArrayList<>();
        ListNode headAux = head;
        int count = 0;
        while(headAux != null){
            lista.add(0,headAux.val);
            headAux = headAux.next;
        }
        int[] numbersArray = lista.stream().mapToInt(i -> i).toArray();
        for(int i=0;i<lista.size()/2;i++){
            if(head.val != numbersArray[i]){
                return false;
            }
            head = head.next;
        }
        return true; 
    }
}